execution-core/src/transfer.rs

```diff
@@ -49,37 +43,22 @@ pub enum Transaction {
 }
 
 impl Transaction {
-    /// Create a new phoenix transaction.
-    #[must_use]
-    pub fn phoenix(payload: PhoenixPayload, proof: impl Into<Vec<u8>>) -> Self {
-        Self::Phoenix(PhoenixTransaction::new(payload, proof))
-    }
-
-    /// Create a new moonlight transaction.
-    #[must_use]
-    pub fn moonlight(
-        payload: MoonlightPayload,
-        signature: AccountSignature,
-    ) -> Self {
-        Self::Moonlight(MoonlightTransaction::new(payload, signature))
-    }
-
```

Comment:
Its unfortunate that these have to be removed, although I understand why. The idea was that we could make a transaction with just the `Transaction` struct. We could wrap the functions in `moonlight` and `phoenix`, but that might defeat the goal.

Reply:
I understand what you mean! And it can actually still be done, we just have to pass more arguments for the construction of each transaction. I will change!